workitem/link/link_repository.go

```diff
@@ -111,6 +133,17 @@ func (r *GormWorkItemLinkRepository) Create(ctx context.Context, sourceID, targe
 	if err := r.ValidateCorrectSourceAndTargetType(ctx, sourceID, targetID, linkTypeID); err != nil {
 		return nil, errs.WithStack(err)
 	}
+	parentExists, err := r.CheckParentExists(ctx, sourceID, targetID, linkTypeID)
+	if err != nil {
+		return nil, err
+	}
+	if parentExists == true {
+		log.Error(ctx, map[string]interface{}{
+			"link_type_id": linkTypeID,
+		}, "unable to create work item link")
```

Comment:
Better! But please, include more properties in the error message helping to read the log better. And also please change the message to include the reason for why we're not able to create a link. Here's an example.

```go
log.Error(ctx, map[string]interface{}{
    "link_type_id": linkTypeID,
    "target_id": targetID,
}, "unable to create work item link because a topology of type \"%s\" only allows one parent to exist and the target already has one", TreeTopology)
```

Now that I'm writing this, I think we need to have a chat on the `CheckParentExists` function. I think it is way to  generic because it looks as if it checks for any parent (even in other topologies such as `tree`), which it doesn't. It only checks for a parent if the link type has a tree topology. Maybe we can make this more obvious.

You know, in other link topologies, a WI can certainly have multiple parents. See [here](https://www.visualstudio.com/en-us/docs/work/reference/link-type-element-reference#link-type-topologies-and-restrictions).